ktor-http/common/src/io/ktor/http/auth/HeaderValueEncoding.kt

```diff
@@ -25,3 +25,7 @@ public enum class HeaderValueEncoding {
      */
     URI_ENCODE
 }
+
+public fun interface HeaderValueEncodingStrategy {
+    public fun chooseEncoding(name: String): HeaderValueEncoding
+}
```

Comment:
If this solution is OK, I will add documentation for this interface.

Reply:
It looks like the target branch should be at least 3.1.0-eap, please check if it is possible to fix without new public api

Reply:
It also might be good to encode directly, so strategy will be an edge case

Reply:
Do you mean adding some `HeaderValueEncoder` which will be responsible for encoding header values?

Reply:
Yep

Reply:
I've managed to fix these bugs without introducing new APIs